File: Backend/CleanArchitecture/CleanArchitecture.Application/Entities/Restaurant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


namespace CleanArchitecture.Core.Entities
{

    public class Restaurant: AuditableBaseEntity
    {
        public string Name { get; set; } // Foreign key to Business
        public string Email { get; set; }
        public string Password { get; set; }
        public string PhoneNumber { get; set; }
        public string StreetInformation { get; set; }
        public string City { get; set; }    
        public string postalCode { get; set; }
        public string Country { get; set; }
        public string StoreType { get; set; }
        public string Picture { get; set; }

        public ICollection<Item> Items { get; set; }
        public ICollection<Reservation> Reservations { get; set; }


        // public LinkedList<Item> Items { get; set; }
        // Other properties
    }
}

File: Frontend/Mobil/app/src/main/java/com/example/leftover/Data/Restaurant.kt
package com.example.leftover.Data

data class AllRestaurant(
    var id: Int,
    var name: String,
    var email: String,
    var password: String,
    var phoneNumber: String,
    var streetInformation: String,
    var city: String,
    var postalCode: String,
    var country: String,
    var storeType: String
)

data class AllAddress(
    val addressDefinition: String,
    val addressTitle: String,
    val addressDirection: String,
    val buildingNo: Int,
    val city: String,
    val floor: Int,
    val id: Int,
    val neighbourhood: String,
    val town: String,
    val userName:String
)

data class RestaurantById(
    val name: String,
    val email: String,
    val password: String,
    val phoneNumber: String,
    val streetInformation: String,
    val city: String,
    val postalCode: String,
    val country: String,
    val storeType: String,
    val picture: String,
    val items: List<Item>,
    val reservations: String,
    val id: Int,
    val createdBy: String,
    val created: String,
    val lastModified: String,
    val lastModifiedBy: String,
)

data class Item(
    val itemName: String,
    val price: Int,
    val picture: String,
    val explanation: String,
    val restaurantId: Int,
    val restaurant: String,
    val reservationItems: String,
    val cartItems: String,
    val id: Int,
    val createdBy: String,
    val created: String,
    val lastModifiedBy: String,
    val lastModified: String,
)

data class SignUp(
    val confirmPassword: String,
    val email: String,
    val firstName: String,
    val lastName: String,
    val password: String,
    val userName: String
)

data class LogIn(
    val email: String,
    val password: String
)

